kernel/src/utilities/single_thread_value.rs

```diff
@@ -0,0 +1,186 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2025.
+
+//! Objects guaranteed to be used by a single thread.
+
+use crate::utilities::cells::OptionalCell;
+
+/// A single-thread store that owns its contents.
+///
+/// This type wraps a value of type `T` by a single thread. Only that thread may
+/// access the value, and that thread may have multiple _shared_ (`&`)
+/// references to the value.
+///
+/// It is [`Sync`] and thus appropriate for static allocations of values that
+/// are not themselves [`Sync`].
+///
+/// # Example
+///
+/// ```
+/// use core::cell::Cell;
+/// use kernel::utilities::single_thread_value::SingleThreadValue;
+///
+/// static FOO: SingleThreadValue<Cell<usize>> = unsafe { SingleThreadValue::new(Cell::new(123)) };
+///
+/// fn main() {
+///   // Provide the SingleThreadValue with a method to inspect the current thread.
+///   FOO.set_chip();
+///
+///   // Attempt to access the value. Passes an Option to the closure.
+///   FOO.with(|foo_option| foo_option.map(|foo| foo.set(foo.get() + 1)));
+///
+///   // Attempt to access the value. Closure is only invoked if permitted.
+///   FOO.with_valid(|foo| foo.set(foo.get() + 2));
+/// }
+/// ```
+///
+/// After creating the [`SingleThreadValue`] and before trying to access the
+/// wrapped value, the [`SingleThreadValue`] must have its [`set_chip()`]
+/// method called. Failing to set the mechanism which can identify threads
+/// will prevent any access to the wrapped value.
+///
+/// # Single-thread Synchronization
+///
+/// It is possible for the same thread to get multiple, shared, references. As a
+/// result, users must use interior mutability (e.g.
+/// [`Cell`](core::cell::Cell), [`MapCell`](tock_cells::map_cell::MapCell), or
+/// [`TakeCell`](tock_cells::take_cell::TakeCell)) to allow obtaining exclusive
+/// mutable access.
+///
+/// # Guaranteeing Single-Thread Access
+///
+/// [`SingleThreadValue`] is safe because it guarantees that the value is only
+/// ever accessed from the same thread that created it. To do this,
```

Comment:
```suggestion
/// ever accessed from the same thread that called `set_chip`. To do this,
```

Reply:
Absolutely, and that "thread" has created and now owns this value. Hm, maybe we should rename set_chip to `create`.